packages/backend/src/tokens/keys.ts

```diff
@@ -0,0 +1,207 @@
+import { API_URL, API_VERSION } from '../constants';
+// DO NOT CHANGE: Runtime needs to be imported as a default export so that we can stub its dependencies with Sinon.js
+// For more information refer to https://sinonjs.org/how-to/stub-dependency/
+import runtime from '../runtime';
+import { callWithRetry } from '../util/callWithRetry';
+import { joinPaths } from '../util/path';
+import { TokenVerificationError, TokenVerificationErrorAction, TokenVerificationErrorReason } from './errors';
+
+type JsonWebKeyWithKid = JsonWebKey & { kid: string };
+
+type JsonWebKeyCache = Record<string, JsonWebKeyWithKid>;
+
+let cache: JsonWebKeyCache = {};
+
+function getFromCache(kid: string) {
+  return cache[kid];
+}
+
+function setInCache(
+  jwk: JsonWebKeyWithKid,
+  jwksCacheTtlInMs: number = 1000 * 60 * 60, // 1 hour
+) {
+  cache[jwk.kid] = jwk;
+
+  if (jwksCacheTtlInMs >= 0) {
+    setTimeout(() => {
+      if (jwk) {
+        delete cache[jwk.kid];
+      } else {
+        cache = {};
+      }
+    }, jwksCacheTtlInMs);
+  }
+}
+
+const LocalJwkKid = 'local';
+
+/**
+ *
+ * Loads a local PEM key usually from process.env and transform it to JsonWebKey format.
+ * The result is also cached on the module level to avoid unnecessary computations in subsequent invocations.
+ *
+ * @param {string} key
+ * @returns {JsonWebKey} key
+ */
+export function loadClerkJWKFromLocal(localKey?: string): JsonWebKey {
+  if (!getFromCache(LocalJwkKid)) {
+    if (!localKey) {
+      throw new TokenVerificationError({
+        action: TokenVerificationErrorAction.SetClerkJWTKey,
+        message: 'Missing local JWK.',
+        reason: TokenVerificationErrorReason.LocalJWKMissing,
+      });
+    }
+
+    // Notice:
+    // Currently Next.js in development mode cannot parse PEM key format, but it can parse JWKs.
+    // This is a simple way to convert our PEM keys to JWKs until the bug is resolved.
```

Comment:
Unrelated to this PR, what would be a good way to track when these 3rd party issues are resolved so we can make the required changes ourselves?

Reply:
Linear ;).